nilearn/_utils/niimg_conversions.py

```diff
@@ -193,22 +223,45 @@ def concat_niimgs(niimgs, dtype=np.float32, accept_4d=False,
                                     niimg.get_affine()))
             if verbose > 0:
                 print("...resampled to first nifti!")
-            
+
             from .. import image  # we avoid a circular import
             niimg = cache(image.resample_img, memory, func_memory_level=2,
                           memory_level=memory_level)(
-                              niimg,
-                              target_affine=target_affine,
-                              target_shape=target_item_shape)
+                                niimg,
+                                target_affine=target_affine,
+                                target_shape=target_item_shape)
             this_data = niimg.get_data()
 
-        data[..., cur_4d_index:cur_4d_index + size] = _to_4d(this_data)
+        data[..., cur_4d_index:cur_4d_index + size] = this_data
         cur_4d_index += size
 
     return new_img_like(first_niimg, data, target_affine)
 
 
-def check_niimgs(niimgs, accept_3d=False, return_iterator=False):
+def _iter_check_niimg_4d(niimgs):
```

Comment:
If somebody asks for an iterator on 4d, this iterator will check for everything during the process. Before that, the code could return niimages with different fov. I like this logic and I think that this could be propagated in other parts of the code.